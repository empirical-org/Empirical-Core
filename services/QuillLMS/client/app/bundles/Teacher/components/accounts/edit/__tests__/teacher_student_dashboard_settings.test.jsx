import * as React from 'react';
import { render, } from '@testing-library/react';

import TeacherStudentDashboardSettings from '../teacher_student_dashboard_settings';

describe('TeacherStudentDashboardSettings component', () => {
  const props = {
    active: false,
    activateSection: jest.fn(),
    deactivateSection: jest.fn(),
    updateTeacherInfo: jest.fn(),
    showStudentsExactScore: true,
  };

  it('should render', () => {
    const { asFragment, } = render(<TeacherStudentDashboardSettings {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

});
