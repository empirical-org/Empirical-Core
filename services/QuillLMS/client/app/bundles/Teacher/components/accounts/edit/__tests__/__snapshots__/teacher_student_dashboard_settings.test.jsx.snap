// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherStudentDashboardSettings component should render 1`] = `
<DocumentFragment>
  <section
    class="user-account-section student-dashboard-settings-section"
  >
    <h1>
      Student dashboard
    </h1>
    <p
      class="student-dashboard-settings-description"
    >
      Control what your classes see on their dashboard. 
      <a
        href="https://support.quill.org/en/articles/9034576-how-do-students-see-their-scores-and-time-spent"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more.
      </a>
    </p>
    <form
      accept-charset="UTF-8"
    >
      <div
        class="fields"
      >
        <div
          class="checkbox-row"
        >
          <button
            aria-label="Unchecked checkbox"
            class="quill-checkbox unselected"
            id="show-exact-score-checkbox"
            type="button"
          />
          <label
            for="show-exact-score-checkbox"
          >
            Show exact score for graded activities
          </label>
        </div>
      </div>
    </form>
  </section>
</DocumentFragment>
`;
